tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/AsynchronousCSTest.java

```diff
@@ -144,205 +112,57 @@ public void testClassLevelAsyncIsNotFinished() {
-        
-        CompletionStage<Connection> resultFuture = clientClass.serviceCS(waitingFuture);
-        complete(waitingFuture);
-
-        Assert.assertTrue(completesQuickly(resultFuture));
-
-
-        try {
-            waitUntilCompleted(resultFuture);
-        }
-        catch (CompletionException e) {
-            handleCompletionException(e);
-        }
+        Task taskToPerform = new ServiceTask();
+        CompletionStage<Task> taskResult = clientClass.serviceCS(taskToPerform);
+        await().untilAsserted(() -> assertTrue(taskResult.toCompletableFuture().isDone()));
     }
 
     /**
      * Test that the callbacks added to the initial stage are executed
      * after the stage returned by the asynchronous method call is completed.
-     * 
-     * The callbacks added inside method invocation must be called first and 
+     * <p>
+     * The callbacks added inside method invocation must be called first and
      * then callbacks added to the result of the call (on the calling thread)
      * must be executed in the order they were added.
      */
     @Test
     public void testAsyncCallbacksChained() {
-        CompletableFuture<Void> waitingFuture = newWaitingFuture();
         StringBuilder executionRecord = new StringBuilder();
-        
-        CompletableFuture<Connection> innerFuture = new CompletableFuture<Connection>();
-        CompletableFuture<Connection> returnedFuture = innerFuture
-                .thenApply(v -> {
-                    executionRecord.append("1");
-                    return v;
-                });
-        CompletionStage<Connection> resultFuture = client
-                .serviceCS(waitingFuture, returnedFuture)
-                .thenApply(v -> {
-                    executionRecord.append("2");
-                    return v;
-                });
-        complete(waitingFuture);
-        resultFuture = resultFuture.thenApply(v -> {
-                    executionRecord.append("3");
-                    return v;
-                });
-
-        Assert.assertFalse(completesQuickly(resultFuture), 
-                "Stage returned by the method isn't completed yet so also the outer stage mustn't be completed");
-        innerFuture.complete(new EmptyConnection());
-        Assert.assertTrue(completesQuickly(resultFuture),
-                "Stage returned by the method is completed so also the outer stage must be completed");
-        Assert.assertEquals(executionRecord.toString(), "123", 
-                "The execution didn't happen in the expected order");
+        Task taskToPerform = new ServiceTask();
+        CompletionStage<Task> taskResult = client.serviceCS(taskToPerform).thenApply(v -> {
+            executionRecord.append("1");
+            return v;
+        }).thenApply(v -> {
+            executionRecord.append("2");
+            return v;
+        }).thenApply(v -> {
+            executionRecord.append("3");
+            return v;
+        });
```

Comment:
This test is not the same as the original test.

The original test has the first thenApply applied to a future which is completed within the asynchronous method, the result of which is then returned.

Please revert this change.